### ISSUE TITLE
execute change set for stack panther-web is ROLLBACK_COMPLETE

### ISSUE THREAD

**Issue by kbroughton** [panther-labs/panther#650]
### Describe the bug
The following command fails.

mage -debug setup:swagger deploy

03:44:15	DEBUG	mage/deploy_template.go:280	deploy: ExecuteChangeSet for stack panther-web is now ROLLBACK_COMPLETE
03:44:15	FATAL	mage/deploy_template.go:290	execute change set for stack panther-web is ROLLBACK_COMPLETE:

In the CloudFormation Events I see

WebApplicationServer | CREATE_FAILED | Unable to assume the service linked role. Please verify that the ECS service linked role exists. (Service: AmazonECS; Status Code: 400; Error Code: InvalidParameterException; Request ID: 150230c7-0d46-4353-ad8d-7afd580f7c47)
-- | -- | --

It seems to be the following IAM role causing problems. Not sure why.

```
  WebApplicationServerExecutionRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
        - Effect: Allow
          Principal:
            Service: ecs-tasks.amazonaws.com
          Action: sts:AssumeRole

```

The first time running dev.sh I tried using the minimal role. It failed because of missing some permissions. Eventually I went back to using my admin role. After hitting this error, I deleted the panther-web stackset and tried again. The error was reproducible every time for me.

### Steps to reproduce
Follow the instructions at https://docs.runpanther.io/quick-start#step-3

./dev.sh
mage -debug setup:swagger deploy

### Expected behavior
completes successfully

### Environment

How are you deploying or using Panther?

commit c7c33378de3765c958cadc1b844eda101e2911aa
Author: Kostas Papageorgiou <kostas@runpanther.io>
Date:   Wed Mar 25 13:52:09 2020 +0100

    Fixing minor stability issues (#564)

**Comment by austinbyers** [comment]
Thanks for the report!

- Not that it should matter, but what region were you deploying to?
- Did it work for you once you deployed with an admin role instead of the deployment role?

**Comment by kbroughton** [comment]
deploying to us-east-1
I switched mid way between CFN creation after the first several fails to my admin creds. However, I did delete the stack each time. I will try a complete clean and redeploy next.

**Comment by austinbyers** [comment]
I haven't been able to reproduce this so far, are you sure you're using the latest deployment role?

From a fresh clone of the repo (commit 29e862ac82, soon to be v1.0.1), I was able to deploy to `us-east-1` using an IAM user with the deployment policy attached from within the docker image:

```bash
./dev.sh
aws sts get-caller-identity # verify user is the deployment user
AWS_REGION=us-east-1 mage setup:swagger deploy
```

The [`WebApplicationServer`](https://github.com/panther-labs/panther/blob/master/deployments/web_server.yml#L55) resource doesn't even reference any IAM roles. @3nvi have you ever seen this error before?

**Comment by 3nvi** [comment]
I've never seen the message `Unable to assume the service linked role. Please verify that the ECS service linked role exists.` Like @austinbyers said, the role should 100% be there before the ECS stack got deployed.

> I deleted the panther-web stackset and tried again
Is there a chance something was left hanging due to manual deletion of a stack?

> I will try a complete clean and redeploy next.
That would surely fix it, cause I feel that something got off sync when deleting the stacks. If you face any issues, I'd be more than willing to help you out. Feel free to send me a PM on our [dedicated Slack channel](https://slack.runpanther.io/)

**Comment by austinbyers** [comment]
Going to go ahead and close this because we've not been able to reproduce for 2 weeks. If you are still experiencing issues after the v1.1 release next week, feel free to open another issue and link back to this one for context.

**Comment by austinbyers** [comment]
Reopening because we finally ran into this ourselves!

**Comment by nhakmiller** [comment]
This issue is caused because by default the `AWSServiceRoleForECS` does not exist, but is required for the execution role to create. This service role should be created by AWS automatically when needed, but it appears not to be working in some cases (perhaps due to permissions configured while deploying). Creating the role by deploying this resource:

```
  ECSServiceLinkedRole:
    Type: AWS::IAM::ServiceLinkedRole
    Properties:
        AWSServiceName: ecs.amazonaws.com
```

in a SEPARATE cloudformation stack before deploying Panther appears to resolve the issue. This will fail to deploy if the role already exists though.

We are looking into a fix for Panther that will make that not required.

**Comment by austinbyers** [comment]
Ok @kbroughton, this is fixed in the [v1.1.1 release!](https://github.com/panther-labs/panther/releases/tag/v1.1.1) Completely teardown an existing deployment, update your deployment role with the latest version, and then try again. Or, if you want to keep your existing deployment, the fix Nick suggested (where you create the service role manually) will also work.

## Root Cause Analysis
The `AWSServiceRoleForECS` is required for the panther-web stack. Normally, ECS will automatically create that service role when you deploy the bootstrap stack, which contains the web cluster. But Panther's deployment role did not have permissions for service roles, this was the bug that was fixed in the release.

When you switched to an admin role, the bootstrap stack (web cluster) had already been created, so ECS didn't try to make the service role that time, even though it had enough permissions that it could have
